src/test/fuzz/connman.cpp

```diff
@@ -23,116 +23,94 @@ void initialize_connman()
 FUZZ_TARGET_INIT(connman, initialize_connman)
 {
     FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
-    CConnman connman{fuzzed_data_provider.ConsumeIntegral<uint64_t>(), fuzzed_data_provider.ConsumeIntegral<uint64_t>()};
-    CAddress random_address;
+    CAddrMan addrman;
+    CConnman connman{fuzzed_data_provider.ConsumeIntegral<uint64_t>(), fuzzed_data_provider.ConsumeIntegral<uint64_t>(), addrman};
     CNetAddr random_netaddr;
     CNode random_node = ConsumeNode(fuzzed_data_provider);
-    CService random_service;
     CSubNet random_subnet;
     std::string random_string;
     while (fuzzed_data_provider.ConsumeBool()) {
-        switch (fuzzed_data_provider.ConsumeIntegralInRange<int>(0, 28)) {
+        switch (fuzzed_data_provider.ConsumeIntegralInRange<int>(0, 23)) {
         case 0:
-            random_address = ConsumeAddress(fuzzed_data_provider);
-            break;
-        case 1:
```

Comment:
nit: may be better to do bitcoin#20828 before for simplicity?
    Merge #20828: fuzz: Introduce CallOneOf helper to replace switch-case

Reply:
i remember backporting https://github.com/bitcoin/bitcoin/pull/20828 and for some reason it caused more problems than it solved, forcing me to manually decrement the cases, which was annoying to work with.